xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/api/Property.java

```diff
@@ -82,18 +85,31 @@ public PropertyClass getPropertyClass()
      */
     public java.lang.Object getValue()
     {
-        BaseProperty baseProperty = getBaseProperty();
+        final BaseProperty<?> baseProperty = getBaseProperty();
+        final XWikiContext context = getXWikiContext();
 
-        com.xpn.xwiki.objects.classes.PropertyClass propertyClass = baseProperty.getPropertyClass(getXWikiContext());
+        com.xpn.xwiki.objects.classes.PropertyClass propertyClass = baseProperty.getPropertyClass(context);
 
         // Avoid dumping password hashes if the user does not have programming rights. This is done only at the
         // API level, so that java code using core classes will still have access, regardless or rights.
         if (propertyClass != null && "Password".equals(propertyClass.getClassType())) {
-            if (!getXWikiContext().getWiki().getRightService().hasProgrammingRights(getXWikiContext())) {
+            if (!context.getWiki().getRightService().hasProgrammingRights(context)) {
+                return null;
+            }
+        }
+
+        if (propertyClass != null && "Email".equals(propertyClass.getClassType())) {
+            if (isObfuscateEmails() && !context.getWiki().getRightService().hasWikiAdminRights(context)) {
```

Comment:
why the admin exception?

Reply:
note that this change doesn't seem related to the `[XWIKI-16138: Email addresses are shown in clear in REST results` issue, is it?

Reply:
this change is not about the REST API, but you asked for it to be included in the JIRA ticket.
at leas that is what I understood from https://jira.xwiki.org/browse/XWIKI-16138?focusedCommentId=101848&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-101848

the admin exception is because "admins should be able to do everything".

Reply:
I raised the fact that we also need to fix it but I didn't say to do it in one issue. You cannot include something that's not related to the issue title ;) Or you need to create different jiras and do a single commit but for several jira issues.

Reply:
> the admin exception is because "admins should be able to do everything".

Do we do that elsewhere (show emails in clear for admins)? I just want to make sure we're consistent in the behavior.

Reply:
Actually we do not do that. Even Admins see the E-mail obfuscated (but of course can go to the object editor for the user to see it there. I guess I should limit the "show emails unobfuscated" to the REST interface, and there only users with programming rights. As requested this will be done in a separate issue.